src/Confluent.Kafka/Consumer.cs

```diff
@@ -269,10 +272,38 @@ public void Unsubscribe()
 
 
         /// <summary>
-        ///     Update the assignment set to a single <paramref name="partition" />.
+        ///     Update the set of assigned partitions (the set of partitions the
+        ///     consumer will consume from) to reflect a <paramref name="rebalanceEvent" />.
+        /// 
+        ///     Note: The newly specified set is the complete set of partitions to
+        ///     consume from. If the consumer is already assigned to a set of partitions,
+        ///     the previous set will be replaced.
+        /// </summary>
+        /// <param name="rebalanceEvent">
+        ///     The rebalance event to use to determine which partitions to assign
+        ///     to. If this is an assignment event, the consumer will be assigned to
+        ///     all partitions in the assignment. If this is a revocation event, the
+        ///     consumer will be unassigned from all partitions.
+        /// </param>
+        public void Assign(RebalanceEvent rebalanceEvent)
+        {
+            if (rebalanceEvent.IsAssignment)
+            {
+                Assign(rebalanceEvent.Partitions);
+            }
+            else
+            {
+                Unassign();
+            }
+        }
+
+        /// <summary>
+        ///     Update the set of assigned partitions (the set of partitions the
```

Comment:
not sure this variant is that useful in practice

Reply:
me either - I'll remove it.